apps/console/src/features/core/utils/http-utils.ts

```diff
@@ -82,14 +82,24 @@ export class HttpUtils {
         */
         const errorConfig: HttpRequestConfig  = error.config as HttpRequestConfig;
         const duration: number = new Date().getTime() - errorConfig?.startTimeInMs;
+        let pathName = null;
 
-        EventPublisher.getInstance().record(
-            new URL(error?.config?.url).pathname,
-            errorConfig?.startTimeInMs,
-            duration,
-            error?.response?.status,
-            false
-        );
+        try {
+            pathName = new URL(error?.config?.url).pathname;
+        } catch(e) {
+            // Add debug logs here one a logger is added.
+            // Tracked here https://github.com/wso2/product-is/issues/11650.
+        }
+
+        if (pathName) {
```

Comment:
In this way, we are not tracking the exceptions of type `AsgardeoAuthException`.

Reply:
The reason for wrapping `new URL(..)` inside the try catch was to prevent the code from breaking when the `error?.config?.url` is undefined.

```js
// throws an exception
new URL(undefined)
```

if this is not handled then the code will not proceed to execute.

Reply:
> if this is not handled then the code will not proceed to execute.

Yes, but by doing this, aren't we skipping tracking exceptions of `AsgardeoAuthException` just because the exception doesn't contain a pathname?

> what happens if start time and duration are passed in as null or undefined to the record function?

AFAIR, `duration` is a mandatory parameter for `trackDepedency` method. If it's passed in as null or undefined, app insights sdk would throw a bad request error.

Reply:
>> Yes, but by doing this, aren't we skipping tracking exceptions of type AsgardeoAuthException just because the exception doesn't contain a pathname?

Oops! You're right. I only got this now. @savindi7 Lets remove the `pathName` check and pass null to the record function.

`record` function should handle the arguments properly.

Reply:
Please refer/review this commit (https://github.com/wso2/identity-apps/pull/3131/commits/3bde53d156bc830ea47881f10aab564ac0b6dae6) and do the needful @pavinduLakshan

Reply:
Done